src/schema/schema_truncate.c

```diff
@@ -41,23 +41,16 @@ __truncate_table(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
  *     Truncate for a tiered data source.
  */
 static int
-__truncate_tiered(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
+__truncate_tiered(WT_SESSION_IMPL *session, const char *uri)
 {
     WT_DECL_RET;
-    WT_TIERED *tiered;
-    u_int i;
 
     WT_RET(__wt_session_get_dhandle(session, uri, NULL, NULL, WT_DHANDLE_EXCLUSIVE));
-    tiered = (WT_TIERED *)session->dhandle;
 
     WT_STAT_DATA_INCR(session, cursor_truncate);
 
-    /* Truncate the tiered entries. */
-    for (i = 0; i < WT_TIERED_MAX_TIERS; i++) {
-        if (tiered->tiers[i].tier == NULL)
-            continue;
-        WT_ERR(__wt_schema_truncate(session, tiered->tiers[i].name, cfg));
-    }
```

Comment:
Note: this truncate algorithm for tiered was changed.  The old way iterates through the tiers, and will end up calling truncate on "file:foo-000000001.wtobj".  But that is not how you get to the tiered Btree, that's basically an empty Btree if you read it off of the disk.  (ASIDE - how to prevent users from opening this?). With an empty btree, the truncate didn't remove any items from the tiered Btree.

Truncate doesn't have any configuration so this function doesn't have a cfg arg anymore (like __truncate_dsrc)